### PR TITLE
Added the route function

### PR DESCRIPTION
**Pull request by ezimuel**
This PR adds a new function `route()` in the view. This function returns a `Zend\Expressive\Router\RouteResult` object in a view. For instance, this can be useful to retrieve the matched route name in a view (or template) using the command `$this->route()->getMatchedRouteName()` or the matched parameters using `$this->route()->getMatchedParams()`.

**NOTE:** This PR needs https://github.com/zendframework/zend-expressive-helpers/pull/71 to be merged.

### CODE REVIEW

File: src/Exception/RuntimeException.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-platesrenderer for the canonical source repository
+ * @copyright Copyright (c) 2016-2017 Zend Technologies USA Inc. (https://www.zend.com)
```
Comment (michalbundyra) [inline src/Exception/RuntimeException.php]: 2019 only, please

File: src/Extension/UrlExtension.php [hunk 2]
```diff
@@ -48,6 +50,22 @@ public function register(Engine $engine) : void
     {
         $engine->registerFunction('url', [$this, 'generateUrl']);
         $engine->registerFunction('serverurl', [$this, 'generateServerUrl']);
+        $engine->registerFunction('route', [$this, 'getRouteResult']);
+    }
+
+    /**
+     * Get the RouteResult instance of UrlHelper
+     *
+     * @return RouteResult
```
Comment (michalbundyra) [inline src/Extension/UrlExtension.php]: it is useless at is not gives more information than function declaration (RTH)

File: src/Extension/UrlExtension.php [hunk 3]
```diff
@@ -48,6 +50,22 @@ public function register(Engine $engine) : void
...
+     */
+    public function getRouteResult() : RouteResult
+    {
+        if (! method_exists($this->urlHelper, 'getRouteResult')) {
+            throw new Exception\RuntimeException(
+                'The helper does not contains the route result. Please update zend-expressive-helpers'
+            );
+        }
```
Comment (michalbundyra) [inline src/Extension/UrlExtension.php]: do we need that exception? Maybe we should bump minimum version of `zend-expressive-helpers` once released?

If needed, we should add `@throws` annotation on the method.

Comment (weierophinney) [reply src/Extension/UrlExtension.php]: I've bumped the minimum supported zend-expressive-helpers version to 5.2 at this time, which allows removing the new exception, as well as the check here.

File: test/Extension/UrlExtensionTest.php [hunk 4]
```diff
@@ -12,9 +12,11 @@
 use League\Plates\Engine;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Prophecy\ProphecyInterface;
+use ReflectionMethod;
```
Comment (michalbundyra) [inline test/Extension/UrlExtensionTest.php]: It seems to be not used

**Pull request merged**